arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedParquetDefinitionLevelReader.java

```diff
@@ -47,35 +46,45 @@ public VectorizedParquetDefinitionLevelReader(
     super(bitWidth, maxDefLevel, readLength, setArrowValidityVector);
   }
 
-  abstract class NumericBaseReader {
+  abstract class BaseReader {
```

Comment:
I have merged `NumericBaseReader` into `BaseReader` to eliminate the duplicate code.

Reply:
Perhaps I can split the refactoring work into a separate PR to make the review much easier.